redis/asyncio/cluster.py

```diff
@@ -234,87 +208,132 @@ def from_url(cls, url: str, **kwargs: Any) -> "RedisCluster":
     def __init__(
         self,
         host: Optional[str] = None,
-        port: int = 6379,
+        port: Union[str, int] = 6379,
+        *,
```

Comment:
@utkarshgupta137 Why do you think it's better to force send arguments as kw only?

Reply:
This starts us down a path split between the two connection objects.
Having that conversation is reasonable, but IMHO not in the 4.x series. It should be part of a general architectural change (i.e maybe sansio and friends).

Reply:
Realistically, I think the issue is that we don't have the same arguments for all (supported) connection types where possible.  I'm okay with adding arguments to sync these, as long as we don't change ordering.

Basically - no backwards compatibility issues.

Reply:
I think the two clients actually should probably inherit from a common base, then work it through there...

Reply:
This is actually how it works in the [rust client](https://docs.rs/redis/latest/redis/struct.Client.html#method.get_connection). Although, in their case, the base class is the client from which you get sync or async connections. I think that is a better experience from a user perspective, but it would be a breaking change and probably needs to be discussed in #2119.